### PR TITLE
Patch OpenCV to recognize MSVC 2022

### PR DESCRIPTION
**Pull request by daniel-riehm**
This PR adds an OpenCV runtime option for MSVC 2022, which is needed for projects which depend on OpenCV to build properly. The patch is applied to all three supported versions of OpenCV.

### CODE REVIEW

**Comment by daniel-riehm** [comment]
jenkins test this please

File: Patches/OpenCV/4.2.0/cmake/templates/OpenCVConfig.root-WIN32.cmake.in [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  elseif(MSVC_VERSION EQUAL 1500)
+    set(OpenCV_RUNTIME vc9)
+  elseif(MSVC_VERSION EQUAL 1600)
+    set(OpenCV_RUNTIME vc10)
+  elseif(MSVC_VERSION EQUAL 1700)
+    set(OpenCV_RUNTIME vc11)
+  elseif(MSVC_VERSION EQUAL 1800)
+    set(OpenCV_RUNTIME vc12)
+  elseif(MSVC_VERSION EQUAL 1900)
+    set(OpenCV_RUNTIME vc14)
```
Comment (mleotta) [inline Patches/OpenCV/4.2.0/cmake/templates/OpenCVConfig.root-WIN32.cmake.in:123]: Did we lose `vc15` here intentionally?

File: Patches/OpenCV/4.5.1/cmake/templates/OpenCVConfig.root-WIN32.cmake.in [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  elseif(MSVC_VERSION EQUAL 1500)
+    set(OpenCV_RUNTIME vc9)
+  elseif(MSVC_VERSION EQUAL 1600)
+    set(OpenCV_RUNTIME vc10)
+  elseif(MSVC_VERSION EQUAL 1700)
+    set(OpenCV_RUNTIME vc11)
+  elseif(MSVC_VERSION EQUAL 1800)
+    set(OpenCV_RUNTIME vc12)
+  elseif(MSVC_VERSION EQUAL 1900)
+    set(OpenCV_RUNTIME vc14)
```
Comment (mleotta) [inline Patches/OpenCV/4.5.1/cmake/templates/OpenCVConfig.root-WIN32.cmake.in:123]: Same here, the previous code had a condition for `vc15`, which is now gone.

**Review by mleotta — CHANGES_REQUESTED** [review]
Mostly looks good, but looking at the diff in the patch it seems there are a couple of place where the code previous supported `vc15` and no longer does.  I'm not sure what version of Visual Studio `vc15` corresponds to.  Is it not needed, or was this a mistake?

**Comment by daniel-riehm** [comment]
jenkins test this please

**Comment by daniel-riehm** [comment]
jenkins test this please

**Review by dstoup — APPROVED** [review]

**Comment by daniel-riehm** [comment]
jenkins test this please

**Comment by mleotta** [comment]
CI errors seem unrelated.  This works for me, let's merge it.

**Pull request merged**